src/tests/ftest/server/replay.py

```diff
@@ -72,9 +77,260 @@ def test_restart(self):
+
+        self.log_step('Verifying data previously written to the dfuse mount point (ior)')
+        self.read_data(ior, container, ppn, dfuse)
+
+        self.log_step('Write additional data to the dfuse mount point (ior)')
+        ior = self.write_data(container, ppn, dfuse)
+
+        self.log.info('Test passed')
+
+    def test_replay_snapshots(self):
+        """Verify POSIX data access after engine restart (DAOS-13010).
+
+        Steps:
+            0) Start 2 DAOS servers with 1 engine per server
+            1) Create a single pool and container in the pool
+            2) Run ior w/ DFS to populate the container with persistent data
+            3) Creating a snapshot (daos container create-snap)
+            4) Repeat steps #2 & #3 three times.
+            5) Verify all three snapshots exist (daos container list-snaps)
+            6) Remove the second snapshot (daos container destroy-snap)
+            7) Verify two snapshots exist (daos container list-snaps)
+            8) Shutdown every engine cleanly (dmg system stop --force)
+            9) Restart each engine (dmg system start)
+            10) Verify all engines have joined (dmg system query)
+            11) Verify two snapshots exist (daos container list-snaps)
+            12) Remove the remaining snapshots (daos container destroy-snap)
+            13) Verify no snapshots exist (daos container list-snaps)
+
+        :avocado: tags=all,daily_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=server,replay
+        :avocado: tags=ReplayTests,test_replay_snapshots
+        """
+        ppn = self.params.get('ppn', '/run/ior_write/*', 1)
+        container = self.create_container()
+
+        snapshots = []
+        for index in range(1, 4):
+            step = join(' of ', index, 3)
+            self.log_step(join(' - ', 'Write data to the container (ior)', step))
+            self.write_data(container, ppn)
+
+            self.log_step(join(' - ', 'Creating a snapshot (daos container create-snap)', step))
+            snapshots.append(container.create_snap()["epoch"])
+
+        self.log_step('Verifying all three snapshots exist (daos container list-snaps)')
+        self.verify_snapshots(container, snapshots)
+
+        self.log_step('Removing the second snapshot (daos container destroy-snap)')
+        container.destroy_snap(epc=snapshots[1])
+        snapshots.pop(1)
+
+        self.log_step('Verifying two snapshots exist (daos container list-snaps)')
+        self.verify_snapshots(container, snapshots)
+
+        self.stop_engines()
+        self.restart_engines(container.pool)
+
+        self.log_step('Verifying two snapshots exist after restart (daos container list-snaps)')
+        self.verify_snapshots(container, snapshots)
+
+        self.log_step('Remove the remaining snapshots (daos container destroy-snap)')
+        for snapshot in snapshots:
+            container.destroy_snap(epc=snapshot)
+        snapshots.clear()
```

Comment:
Just FYI, to avoid a manual `clear()`
```suggestion
        while snapshots:
            container.destroy_snap(epc=snapshots.pop(0))
```

Reply:
Done